src/HotChocolate/Spatial/src/Types.Spatial/Input/PointInputObject.cs

```diff
@@ -0,0 +1,120 @@
+using System.Collections.Generic;
+using System.Linq;
+using HotChocolate.Configuration;
+using HotChocolate.Language;
+using HotChocolate.Types;
+using HotChocolate.Types.Descriptors;
+using HotChocolate.Types.Descriptors.Definitions;
+using HotChocolate.Utilities.Serialization;
+using NetTopologySuite;
+using NetTopologySuite.Geometries;
+
+namespace Types.Spatial.Input
+{
+    public class PointInputObject : InputObjectType<Point>
+    {
+        private const string _longitude = "x";
+        private const string _latitude = "y";
+        private const string _srid = "srid";
+
+        private IInputField _longitudeField;
+        private IInputField _latitudeField;
+        private IInputField _sridField;
+
+        public PointInputObject()
+        {
+            _longitudeField = default!;
+            _latitudeField = default!;
+            _sridField = default!;
+        }
+
+        protected override void Configure(IInputObjectTypeDescriptor<Point> descriptor)
+        {
+            descriptor.BindFieldsExplicitly();
+
+            // required fields
+            descriptor.Field(_longitude)
+                .Type<NonNullType<FloatType>>()
+                .Description("X or Longitude");
+
+            descriptor.Field(_latitude)
+                .Type<NonNullType<FloatType>>()
+                .Description("Y or Latitude");
+
+            // optional fields
+            descriptor.Field("srid")
+                .Type<IntType>()
+                .Description("Spatial Reference System Identifier");
+        }
+
+        public override object? ParseLiteral(IValueNode literal)
+        {
+            if (literal is NullValueNode)
+            {
+                return null;
+            }
+
+            if (!(literal is ObjectValueNode obj) || obj.Fields.Count < 2)
+                throw new InputObjectSerializationException(
```

Comment:
we always have brackets

Reply:
this should go in the style guide and editorconfig ;)